src/core/server/logging/appenders/rolling_file/rolling_file_context.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { statSync } from 'fs';
+
+/**
+ * Context shared between the rolling file manager, policy and strategy.
+ */
+export class RollingFileContext {
+  constructor(public readonly filePath: string) {}
+  /**
+   * The size of the currently opened file.
+   */
+  public currentFileSize: number = 0;
+  /**
+   * The time the currently opened file was created.
+   */
+  public currentFileTime: number = 0;
+
+  public refreshFileInfo() {
+    try {
+      const { birthtime, size } = statSync(this.filePath);
+      this.currentFileTime = birthtime.getTime();
+      this.currentFileSize = size;
+    } catch (e) {
+      this.currentFileTime = Date.now();
```

Comment:
Ideally, we would only catch `ENOENT` (or such) errors. But that's kinda the first time we got a logger/appender implementation complex enough to potentially throw errors, so I'm not really sure how we should handle failures internal to the logger system or its appenders, so for safety I used a catch-all instead.

Reply:
Good point, will add a `ENOENT` check and logs the error.

Remaining question (same as in https://github.com/elastic/kibana/pull/84735#discussion_r535490305) is, should we:
1. use `console.log`
2. use `console.error`
3. create a `ConsoleAppender` in the `RollingFileAppender` and use it internally for all error logging (we would use the same `layout` as the one provided to the rolling file appender). In that case, it will logs in `stdout` as if we were using `console.log`

wdyt?

Reply:
The 3rd option looks fine. However, I'd expect that it uses the default `layout` that a user specifies for the `root` context.
If the 3rd option is too hard to implement, the 2nd one is well enough.